source/RoslynAnalyzers/Testing/Integration/IntegrationTestContainerAnalyzer.cs

```diff
@@ -34,27 +34,122 @@ internal override void AnalyzeCompilation(INamedTypeSymbol classSymbol, SymbolAn
                 context.ReportDiagnostic(Diagnostic.Create(Descriptors.Oct2005DoNotNestIntegrationTestContainerClasses, classSymbol.Locations.First()));
             }
 
-            var memberSymbolsThatAreNotTypeOrMethodDefinitions = classSymbol
-                .GetMembers()
-                .ExceptOfType<ITypeSymbol>()
-                .ExceptOfType<IMethodSymbol>()
-                .ExceptImplicitlyDeclared();
-
-            foreach (var symbol in memberSymbolsThatAreNotTypeOrMethodDefinitions)
+            foreach (var symbol in classSymbol.GetMembers())
```

Comment:
I'm being very paranoid about performance here (analyzers run on every keystroke after all) and LINQ is known to be bad for performance. A couple of LINQ queries would have been fine, but to hit all the cases this checks for now I would have needed to do maybe 5 or 6 and I wasn't feeling comfortable with that.

Reply:
Line 22 already limits the scope drastically, so this part will only get hit 1000's of times